test/integration/test-all.sh

```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
+
+set -e
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+cd "${DIR}"
+
+for testdir in $(find . -type f -name setup); do
```

Comment:
Using `setup` as a sentinel for "this is a test" is a bit fishy to me. 

Maybe the directory should look like 

```
test/integration/suites/<suite name>/
```

and you could do a single depth find in that directory, or a `for dir in GLOB` style thing.

Also, consider giving the setup scripts in the "suite" ROOTDIR and TESTDIR as positionals, instead of environment variables. I think it will read a bit clearer.

Reply:
That works for me. Mind if I tackle that in a follow up (coming soon)?